ui/app/AppLayouts/Wallet/popups/swap/SwapModalAdaptor.qml

```diff
@@ -130,6 +134,21 @@ QObject {
                 formattedBalance: "0 %1".arg(root.fromToken.symbol)
             }
         }
+
+        property string routeErrorMessage: {
```

Comment:
this could `swapOutputData.errDescription` itself?  I think its very confusing to track where the error messages come from currently. Part of logic is in SwapModal and some here. Can we combine these and have a single function and define all error messages? and then simply use that in the ErrorTag to display it?

Reply:
Yeah, there's two sources for errors currently
- We have some errors related to internal checks, that don't need to wait for a proposal response (amount entered greater than balance)
- Errors related to the proposal response

I can try to move everything we need to the Adaptor so the message and error states get calculated in a single spot.